### PR TITLE
[SPARK-27787][ML] Eliminate uncessary job to compute SSreg

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
Eliminate uncessary job to compute SSreg
Compute SSreg based on the summary of predictions

## How was this patch tested?
existing tests

### CODE REVIEW

**Comment by zhengruifeng** [comment]
retest this please

File: mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala [hunk 1]
```diff
@@ -60,29 +60,23 @@ class RegressionMetrics @Since("2.0.0") (
...
   private lazy val SStot = summary.variance(0) * (summary.weightSum - 1)
-  private lazy val SSreg = {
-    val yMean = summary.mean(0)
-    predictionAndObservations.map {
-      case (prediction: Double, _: Double, weight: Double) =>
-        math.pow(prediction - yMean, 2) * weight
-      case (prediction: Double, _: Double) => math.pow(prediction - yMean, 2)
-    }.sum()
-  }
+  private lazy val SSreg = math.pow(summary.normL2(2), 2) +
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala]: It took me a minute but yes I think this is correct.
It feels like this code would be more straightforward if the summarizer would just expose a sum and squaredL2Norm method, but, maybe for another time.
A few comments about the computations here wouldn't hurt.

**Review by srowen — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**